### PR TITLE
[pvr] Add recordings undelete support

### PR DESCRIPTION
**Pull request by AlwinEsch**
With this change is it possible to access deleted recordings on backend to undelete or delete permanent.
If deleted recordings are accessible a new button is shown on the recordings window to switch there in the list.

The addon header become 3 other functions and 2 with a new value.

GetRecordingsAmount = Have now a value to ask about deleted
GetRecordings = the same
DeleteRecording = Must delete everything, to tray and possible to delete complete
UndeleteRecording = new, to undelete given recording
DeleteAllRecordingsFromTrash = new, to make trash empty

On the file item it can be identified as recording and also with 2 new function to know whether it is available or not.

Have a change for the VDR VNSI here:
Addon: https://github.com/AlwinEsch/kodi-pvr-addons/tree/vnsi-undelete-recordings
Backend Plugin: https://github.com/AlwinEsch/vdr-plugin-vnsiserver/tree/add-undelete-support

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Hello @xhaggi and @opdenkamp,
for a for me unknown reason the other request #6261 was closed after a update and not possible to reopen.

Have changed to the use of shared ptr and tried to use a separate class for deleted ones but seen there a lot of stupid double things and switching map on functions (<code>PVR_RECORDINGMAP *map = isDeleted ? &m_recordingsDeleted : &m_recordingsActive;</code>). Is it OK for you to leave at in one map?

**Comment by xhaggi** [comment]
@AlwinEsch you wiped out all your changes (commits) in the PR. That's the reason why you can't re-open it.

**Comment by xhaggi** [comment]
one general question: is it possible to playback a deleted recording?

**Comment by AlwinEsch** [comment]
No, is inhibited. Want to have for all addons the same. I think it become confusing if on one addon it is possible and on other not.

**Comment by opdenkamp** [comment]
other than the magic numbers from the language file without comments, it's +1

**Comment by AlwinEsch** [comment]
Is updated. Have added to the string id's the name and added select dialog if more as one addon is present with reproducable recordings.

Also created pull requests to VDR vnsi plugin and PVR addons.

File: language/English/strings.po [hunk 1]
```diff
@@ -8838,7 +8850,43 @@ msgctxt "#19289"
...
+#. Used on yes no dialog
+#: xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
+msgctxt "#19293"
+msgid "Delete all recordings permanently?"
+msgstr ""
+
+#. Used on yes no dialog header
+#: xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
+msgctxt "#19294"
+msgid "Confirm undelete"
```
Comment (da-anda) [inline language/English/strings.po]: from a usability POV, do we need to confirm an undelete action? I mean this action does no harm at all and I don't know of any OS that asks for confirmation to undelete something.

Comment (AlwinEsch) [inline language/English/strings.po]: Can be removed, is a bit to much ;-). Thought on beginto make sure thats right before on addon some work becomes performed.

Comment (jjd-uk) [inline language/English/strings.po]: Agreed I think it's an unnecessary extra step.

File: language/English/strings.po [hunk 2]
```diff
@@ -8342,7 +8346,11 @@ msgctxt "#19178"
 msgid "Show channel information when switching channels"
 msgstr ""
 
-#empty string with id 19179
+#. Used as extension header on recordings window
+#: xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
+msgctxt "#19179"
+msgid "Deleted recordings trash"
```
Comment (jjd-uk) [inline language/English/strings.po]: Just been checking the usage of English for the strings.po additions and wasn't sure about this one, could you confirm where & how it's used? a screenshot would be great if you have the time.

Comment (AlwinEsch) [inline language/English/strings.po]: No problem give me few minutes to compile

Comment (jjd-uk) [inline language/English/strings.po]: Was just wondering if we could simply go with "Deleted recordings", so is the "trash" really necessary.

Comment (AlwinEsch) [inline language/English/strings.po]: Was added from me to show that is possible to undelete. If it become removed the also Recordings.

Is currently:
"Recordings - Deleted recordings trash"
if become changed then I use.
"Recordings - Deleted"

**Comment by AlwinEsch** [comment]
Here a screenshot.
bildschirmfoto2

File: language/English/strings.po (hunk 2 above)
Comment (jjd-uk) [inline language/English/strings.po]: Thanks, in that case "Deleted" as the window name I think would be enough as you have the context of "TV - Recordings - Deleted" displayed.

@da-anda for a 2nd opinion

Comment (da-anda) [inline language/English/strings.po]: yes, "Deleted" should be enough. We could go with something like "Recycler" to be in line with some operating systems, but "Deleted" is probably better in this particular case

**Comment by AlwinEsch** [comment]
Is updated, thanks for your comments.

**Comment by AlwinEsch** [comment]
Is transfered to #6367 and this becomes closed.

**Pull request closed without merging**